src/components/UserAccess/UserAccessForm/form-utils.ts

```diff
@@ -97,14 +97,7 @@ export const deleteRB = async (roleBinding: RoleBinding, dryRun?: boolean): Prom
     },
   };
 
-  // K8sQueryDeleteResource enjoys the k8sDeleteResource to delete resources.
-  // However, based on the design of k8sDeleteResource, 'dryRun' just affect the
-  // query parameter but would not affect commonFetchJSON.delete.
-  // That is to say, with 'true' dryRun, the resource would be also deleted.
-  // If so, we have to skip the redeletion when dryRun is false.
-  // To do: let us improve the function when the K8sQueryDeleteResource
-  // works well with dryRun.
-  dryRun ? await K8sQueryDeleteResource(queryOptions) : void 0;
```

Comment:
If dry run is deleting resources, the we should not call delete resource method in dry run phase.